AO3-6235 Allow policy_and_abuse admins to toggle invite requests in settings

dismayonnaise opened the pull request:
# Pull Request Checklist

<!-- Mark steps in the checklist as complete by changing `[ ]` to `[X]` -->
* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
  * ~~Tests were not added because a) this was a small change that was easily manually tested, and b) I couldn't find tests covering other permissions for `policy_and_abuse` archive settings. That being said, if I missed tests or if I should add tests, please let me know and I'll be happy to add them.~~
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6235

## Purpose

This allows admins with the `policy_and_abuse` role to toggle invitation requests via Archive Settings.

## Testing Instructions

The JIRA ticket already has test instructions. However, additional testing can cover edge and negative cases:

* For an admin with additional roles including `policy_and_abuse`, test if you can toggle invitation requests (ex: an admin that has both the `policy_and_abuse` and `support` role).
* For an admin without `policy_and_abuse` role, check that you cannot toggle invitation requests.

## Credit

dismayonnaise (she/her)

💖 💖 Thank you!! 💖 💖

sarken commented:
Hi, dismayonnaise!

Thank you so much for this pull request and additional testing instructions! Someone will be along to review it soon.

In the meantime, I've assigned the Jira issue to you and set its status to In Review, so no one will mistakenly create a duplicate pull request. I've also given your account permission to comment on, assign, and transition issues in the future.

Thanks again for contributing! If you have any questions, you can contact us at otw-coders@transformativeworks.org.

dismayonnaise commented:
Thank you for finding that test and the awesome feedback @weeklies!! I went ahead and updated the test to include the `request_invite_enabled` field.

I also refactored the test in my last commit. The reason I refactored the test is because the test parameter format didn't quite match what the frontend form is submitting. The form is submitting `0`/`1` for boolean values instead of `false`/`true`. I updated the test to use `0`/`1` for boolean values instead.  That being said, my refactoring might be overkill. Let me know what you think. If it's too much/not good/etc etc, I'm more than happy to revert it.

Thank you again for checking out my PR 😄 I really appreciate it!!

weeklies reviewed (approved):
Looks good to me, thank you.
